File: ZestincLauncher/app/src/main/java/com/example/zestinc/zestinclauncher/AppsAdapter.java
package com.example.zestinc.zestinclauncher;

import android.app.Activity;
import android.content.pm.ResolveInfo;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.GridView;
import android.widget.ImageView;

import java.util.List;

public class AppsAdapter extends BaseAdapter {

    private List<ResolveInfo> mApps;
    private Activity mActivity;

    AppsAdapter(List<ResolveInfo> apps, Activity activity){
        mApps = apps;
        mActivity = activity;
    }

    @Override
    public int getCount() {
        return mApps.size();
    }

    @Override
    public Object getItem(int i) {
        return mApps.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }


    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        ImageView iv;

        if(view == null){
            iv = new ImageView(mActivity);
            iv.setScaleType(ImageView.ScaleType.FIT_CENTER);
            int iconSize = mActivity.getResources().getDimensionPixelSize(R.dimen.app_icon_size);
            iv.setLayoutParams(new GridView.LayoutParams(iconSize, iconSize));
        } else {
            iv = (ImageView) view;
        }
        ResolveInfo info = mApps.get(i);
        iv.setImageDrawable(info.activityInfo.loadIcon(mActivity.getPackageManager()));
        return iv;
    }
}

File: ZestincLauncher/app/src/main/java/com/example/zestinc/zestinclauncher/AppsAdapter.kt
package com.example.zestinc.zestinclauncher

import android.app.Activity
import android.content.pm.ResolveInfo
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.GridView
import android.widget.ImageView

class AppsAdapter internal constructor(private val mApps: List<ResolveInfo>, private val mActivity: Activity) : BaseAdapter() {

    override fun getCount(): Int {
        return mApps.size
    }

    override fun getItem(i: Int): Any {
        return mApps[i]
    }

    override fun getItemId(i: Int): Long {
        return i.toLong()
    }


    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
        val iv: ImageView

        if (view == null) {
            iv = ImageView(mActivity)
            iv.scaleType = ImageView.ScaleType.FIT_CENTER
            val iconSize = mActivity.resources.getDimensionPixelSize(R.dimen.app_icon_size)
            iv.layoutParams = ViewGroup.LayoutParams(iconSize, iconSize)
        } else {
            iv = view as ImageView
        }
        val info = mApps[i]
        iv.setImageDrawable(info.activityInfo.loadIcon(mActivity.packageManager))
        return iv
    }
}